src/aggregate/aggregate_exec.c

```diff
@@ -521,7 +521,7 @@ static void sendChunk_Resp2(AREQ *req, RedisModule_Reply *reply, size_t limit,
 
     if (resultsLen != REDISMODULE_POSTPONED_ARRAY_LEN && rc == RS_RESULT_OK && resultsLen != nelem) {
       RedisModule_Log(RSDummyContext, "warning", "Failed to predict the number of replied results. Prediction=%ld, actual_number=%ld.", resultsLen, nelem);
-      RS_LOG_ASSERT(0, "Precalculated number of replies must be equal to actual number");
+      RS_ABORT("Precalculated number of replies must be equal to actual number");
```

Comment:
Could be dropped since it is next to a `RedisModule_Log`?

Reply:
How does the log achieve the effect of an assertion?

Reply:
Dropped